flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/FlinkService.java

```diff
@@ -129,6 +129,16 @@ public boolean isJobManagerPortReady(Configuration config) {
         return false;
     }
 
+    public boolean isJobManagerServing(Configuration config) {
+        try (ClusterClient<String> clusterClient = getClusterClient(config)) {
+            clusterClient.listJobs().get(10, TimeUnit.SECONDS);
+            return true;
+        } catch (Exception ignored) {
+        }
```

Comment:
Does this need to log the exception for listing the jobs of the cluster client?

Reply:
The `observeJmDeployment()` method will output log WARN msg when the rest service is not working. 
Adding it here seems to be redundant?

Reply:
I don't think this is correct and we need to add test coverage. We need to cover this: https://github.com/apache/flink-kubernetes-operator/pull/56/files#diff-1d1ff01af8cbc1561adcc4e62e4f8f4280c7aa8c7b3f0624eb5f27f09aa9e96fR73

Reply:
It might be better to leave this logic in the observer.

Reply:
@tweise I dont really understand your comment, it seems to refer to changes you introduced in your PR.

Reply:
It would be very nice of @bgeng777 to add test coverage to guard the current behaviour but it's not easy to understand all these corner cases that should have been covered already by tests. Let's try to test and review this carefully and we should then take the time to add the missing test coverage for the older features.

Reply:
I will try to cover the listJobs error scenario in a separate PR soon.